src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,217 @@
+                + "where c.id = ? AND c.is_deleted = false;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                        = connection.prepareStatement(selectRequest)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                car = parseCarWithManufacturerFromResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get car by id " + id, e);
+        }
+        if (car != null) {
+            car.setDrivers(getDriversForCar(id));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String selectAllRequest = "SELECT cars.id, model, "
+                + "manufacturers.id, manufacturers.name, manufacturers.country "
+                + "FROM cars "
+                + "JOIN manufacturers "
+                + "ON cars.manufacturer_id = manufacturers.id + "
+                + "WHERE cars.is_delete = false;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllStatement
+                        = connection.prepareStatement(selectAllRequest)) {
+            ResultSet resultSet = getAllStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(parseCarWithManufacturerFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get all cars from DB", e);
+        }
+        for (Car car : cars) {
+            car.setDrivers(getDriversForCar(car.getId()));
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateRequest = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_delete = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateStatement
+                        = connection.prepareStatement(updateRequest)) {
+            updateStatement.setString(1, car.getModel());
+            updateStatement.setLong(2, car.getManufacturer().getId());
+            updateStatement.setLong(3, car.getId());
+            updateStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update a car " + car, e);
+        }
+        addDriverToCar(car);
+        removeDriverFromCar(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long carId) {
+        String deleteRequest = "UPDATE cars SET is_deleted = true WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement softDeleteCarStatement
```

Comment:
```suggestion
                PreparedStatement statement
```
same thing as with query namings - obvious context